specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2021-01-01/ResourceHealth.json

```diff
@@ -1164,6 +1236,173 @@
         }
       }
     },
+    "potentialOutagesGetResult": {
+      "description": "The Get PotentialOutages operation response.",
+      "type": "object",
+      "$ref": "#/definitions/potentialoutage",
+      "readOnly": true
```

Comment:
Question here, was trying to find documentation around this but are you sure that setting readOnly at this level will inherit all the way to all your "potentialoutage" properties? I think the previous comment was to actually add readOnly to all properties within your "potentialoutage" definition.

Reply:
it appears to be swagger version dependent: https://stackoverflow.com/questions/51402156/how-to-declare-a-ref-property-as-readonly-in-openapi-swagger. I made the change to address the ambiguity here.

Reply:
@filizt found an example and it seems that readOnly is not inherited, your only option is to add readOnly to every single property. Modeling a property as readOnly indicates the SDK generation to expose a public getter but a private setter, otherwise both are going to be publics, this is ok if you are reusing your definition in a PUT and GET